File: 백준/Silver/25206. 너의 평점은/너의 평점은.cc
#include <string>
#include <iostream>

using namespace std;

int main() {
    float grade;
    float score;
    string s;
    double avg;
    double scoreSum = 0;
    double gradeSum = 0;
    string major;
    for (int i = 0; i < 20; i++) {
        cin >> major >> grade >> s;
        if (s == "A+") score = 4.5;
        else if (s == "A0") score = 4.0;
        else if (s == "B+") score = 3.5;
        else if (s == "B0") score = 3.0;
        else if (s == "C+") score = 2.5;
        else if (s == "C0") score = 2.0;
        else if (s == "D+") score = 1.5;
        else if (s == "D0") score = 1.0;
        else if (s == "F") score = 0.0;
        else if (s == "P") {
            score = 0.0;
            grade = 0;
        }
        
        scoreSum += grade * score;
        gradeSum += grade;
    }
    
    avg = scoreSum / gradeSum;
    cout << fixed;
	cout.precision(6);
	cout << avg;
}

File: 백준/Silver/25206. 너의 평점은/너의 평점은.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        double score = 0;
        double scoreSum = 0;
        double creditSum = 0;
        double avg = 0;

        for (int i = 0; i < 20; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            String major = st.nextToken();
            float credit = Float.parseFloat(st.nextToken());
            String s = st.nextToken();
            if (s.equals("A+")) score = 4.5;
            else if (s.equals("A0")) score = 4.0;
            else if (s.equals("B+")) score = 3.5;
            else if (s.equals("B0")) score = 3.0;
            else if (s.equals("C+")) score = 2.5;
            else if (s.equals("C0")) score = 2.0;
            else if (s.equals("D+")) score = 1.5;
            else if (s.equals("D0")) score = 1.0;
            else if (s.equals("F")) score = 0.0;
            else if (s.equals("P")) {
                score = 0.0;
                credit = 0;
            }
            scoreSum += credit * score;
            creditSum += credit;
        }

        avg = scoreSum / creditSum;
        System.out.println(avg);
    }
}